Build workflow

nirs opened the pull request:
Testing build workflow.

medyagh reviewed (approved):
approved and should merge but still didnt fix the error
```
b02dd694b59c: Pull complete
58fe37f35d87: Pull complete
Digest: sha256:a0655d9b3aea65de70e5391abe468634da3c7d32a234902661eff6fce7bbd669
Status: Downloaded newer image for quay.io/nirsof/xcgo:jammy-v2
GOOS=darwin CGO_ENABLED=1 go build \
	-ldflags="-X k8s.io/minikube/pkg/drivers/hyperkit.version=v1.36.0 -X k8s.io/minikube/pkg/drivers/hyperkit.gitCommitID="fbee19af8ce7bbc0ff9956b9cfce665fa754506d""   \
	-o out/docker-machine-driver-hyperkit k8s.io/minikube/cmd/drivers/hyperkit
+ failed=2
```

nirs committed:
Add Makefile

Makes it easier to build the image with the right image url. This will
also used in the CI.

nirs committed:
Add Build workflow

Build the image for every PR, when pushing to master, or per request.

We need another workflow for releasing an image with a tag.
